src/Functions/FunctionsCoding.h

```diff
@@ -1326,6 +1326,395 @@ class FunctionUnhex : public IFunction
+            {
+                size_t new_offset = prev_offset + n;
+
+                executeOneString(&in_vec[prev_offset], &in_vec[new_offset], pos);
+
+                out_offsets[i] = pos - begin;
+                prev_offset = new_offset;
+            }
+
+            if (!out_offsets.empty() && out_offsets.back() != out_vec.size())
+                throw Exception("Column size mismatch (internal logical error)", ErrorCodes::LOGICAL_ERROR);
+
+            col_res = std::move(col_str);
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t /*input_rows_count*/) const override
+    {
+        const IColumn * column = arguments[0].column.get();
+        ColumnPtr res_column;
+
+        if (tryExecuteUInt<UInt8>(column, res_column) || tryExecuteUInt<UInt16>(column, res_column)
+            || tryExecuteUInt<UInt32>(column, res_column) || tryExecuteUInt<UInt64>(column, res_column)
+            || tryExecuteString(column, res_column) || tryExecuteFixedString(column, res_column)
+            || tryExecuteFloat<Float32>(column, res_column) || tryExecuteFloat<Float64>(column, res_column)
+            || tryExecuteDecimal<Decimal32>(column, res_column) || tryExecuteDecimal<Decimal64>(column, res_column)
+            || tryExecuteDecimal<Decimal128>(column, res_column))
+            return res_column;
+
+        throw Exception(
+            "Illegal column " + arguments[0].column->getName() + " of argument of function " + getName(), ErrorCodes::ILLEGAL_COLUMN);
+    }
+};
+
+class FunctionUnbin : public IFunction
+{
+public:
+    static constexpr auto name = "unbin";
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionUnbin>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+    bool isInjective(const ColumnsWithTypeAndName &) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (!isString(arguments[0]))
+            throw Exception(
+                "Illegal type " + arguments[0]->getName() + " of argument of function " + getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    static void unbinOne(const char * pos, const char * end, char *& out)
+    {
+        uint8_t left = 0;
+        for (int left_cnt = (end - pos) & 7; left_cnt > 0; --left_cnt)
+        {
+            left = left << 1;
+            if (*pos == '1')
+            {
+                left += 1;
+            }
+            ++pos;
+        }
+        if (0 != left)
+        {
+            *out = left;
+            ++out;
+        }
+
+        while (end - pos != 0)
+        {
+            int c = 0;
```

Comment:
Should it be byte, not int?
```suggestion
            UInt8 c = 0;
```